### DISCUSSION TITLE
如果获取下载进度以及下载完成事件

### DISCUSSION THREAD

**Discussion by yangqiong** [bytedance/xgplayer#878] [Q&A] [answered]
**您使用的西瓜播放器版本是多少？ What version of xgplayer are you using?**
v2.31.2

**您使用的操作系统和浏览器分别是？ What OS and browser are you using?**
Mac Chrome

**如何复现问题？ What did you do?**
点击下载按钮

**您期望的播放器正常行为是？ What did you expect to happen?**
能够获取下载进度以及下载完成事件

**实际播放器的表现是？ What actually happened?**
无反应

**可填写您所在的公司和相关产品业务，方便我们提供更好的技术支持 You can write your company and product which uses xgplayer,  for helping us provide better technical support.**

**Comment by gemxx** [comment, marked as answer]
download 插件采用的开源 [downloadjs](https://github.com/rndme/download) 实现的下载功能，由于该开源实现的原理，无法提供统一的下载进度事件，因此xgplayer也不提供相关事件。如果有强烈业务需求，可以参考 [downloadjs](https://github.com/rndme/download)  xmlHttpRequest 逻辑实现一个[自定义](https://v2.h5player.bytedance.com/plugins/#%E8%87%AA%E5%AE%9A%E4%B9%89%E6%8F%92%E4%BB%B6)的下载插件
